docs/workforce-data-import/monthly-process.md

```diff
@@ -0,0 +1,157 @@
+
+- Check files are available:
+  - 2 files should be ready to download on the 25th of each month from an agreed secure location.  
+  The files will be named Workforce-Dataset-1-*YYYYMM*25-1329.csv and Workforce-Dataset-2-*YYYYMM*25-1329.csv where *YYYYMM* is the year and month of the extract date.  
+  - If this is not the case then email the contact at TPS to chase when it will be available.
+
+- Ensure files are in CSV format:
+  - The files contain 7 header lines which include the word **RESTRICTED** which need to be removed to truly make the file a **CSV**.  
+  - Open each file in a text editor[^1] such as **Notepad** or **Visual Studio Code**.
+  - Delete the additional header lines above the field names.
+  - Save the file.
+
+### 2. Upload files to Azure blob storage
+
+- Request access to the **s189 TRA production PIM** group at Home -> Privileged Identity Management -> My Roles -> Groups.
+- Navigate to the blob storage container where the files should be uploaded at Home -> Storage accounts -> s189p01trspdsa -> Containers -> tps-extracts.
+- Upload both files to a virtual folder called **Pending**.
+- Delete the files from all other temporary locations other than the **Pending** folder.
+
+### 3. Trigger the import job in Hangfire
+
+- Navigate to the TRS Hangfire dashboard at https://trs-production-ui.teacherservices.cloud/_hangfire (you will need to be set as an Administrator in TRS).
+- Click **Recurring Jobs**.
+- Click the checkbox next to **ImportTpsCsvExtractFileJob**.
+- Click the **Trigger now** button.
+- Click on **Jobs** and monitor until there is no longer anything *Processing* (this can take around 20 minutes).
+- Repeat above steps in order for the 2nd file to get processed.
+
+### 4. Check the TRS database for any issues with the import
+
+#### Connect to a Kubernetes pod
+
+- Request access to the **s189 TRA production PIM** group at Home -> Privileged Identity Management -> My Roles -> Groups (if not already done previously).
+- Start a **Powershell** session in a tool such as **Visual Studio Code** or **Windows Terminal**.
+- Login to Azure using Azure CLI:
+  - Run `az login` if not logged in previously. This will take you to the Azure portal to authenticate the CLI.
+- Install Kubernetes tools **kubectl** and **kubelogin**:
+  - Run `az aks install-cli`.
+- Set the current Azure account / subscription:
+  - Run `az account set --subscription s189-teacher-services-cloud-production`.
+- Setup local Kubernetes config:
+  - Run `az aks get-credentials --overwrite-existing -g s189p01-tsc-pd-rg -n s189p01-tsc-production-aks`.
+  - Run `kubelogin convert-kubeconfig -l azurecli`.
+- List the Kubernetes pods in the production namespace:
+  - Run `kubectl get pods -n tra-production`.
+- Choose a Kubernetes pod starting with the prefix **trs-production** and start an interactive shell:
+  - Run `kubectl exec --stdin --tty trs-production-<appropriate pod> -n tra-production -- /bin/ash`.
```

Comment:
You can use the deployment name instead of a specific pod:
`kubectl exec --stdin --tty deploy/trs-production-worker -n tra-production -- /bin/ash`